src/consensus/encode.rs

```diff
@@ -597,6 +644,33 @@ pub(crate) fn consensus_encode_with_size<S: io::Write>(data: &[u8], mut s: S) ->
 }
 
 
+struct ReadBytesFromFiniteReaderOpts {
+    len: usize,
+    chunk_size: usize,
+}
+
+/// Read `len` bytes from reader, where `len` could potentially be malicious
+///
+/// This function relies on `reader` being bound in amount of data
```

Comment:
```suggestion
/// This function relies on reader being bound in amount of data
```
Since the argument is `d` not `reader`.